component/src/main/java/org/wso2/extension/siddhi/execution/geo/GeoCoordinateStreamFunctionProcessor.java

```diff
@@ -0,0 +1,168 @@
+    private static GeoCoordinateResolver geoCoordinateResolverImpl;
+    private static final String DEFAULT_GEOCOORDINATE_RESOLVER_CLASSNAME =
+            "org.wso2.extension.siddhi.execution.geo.internal.impl.APIBasedGeoCoordinateResolver";
+
+    @Override
+    protected Object[] process(Object[] data) {
+        throw new IllegalStateException("geocoordinate cannot execute for single data ");
+    }
+
+    @Override
+    protected Object[] process(Object data) {
+        String ip = data.toString();
+        GeoCoordinate geoCoordinate = geoCoordinateResolverImpl.getGeoCoordinateInfo(ip);
+        return new Object[]{geoCoordinate.getLatitude(), geoCoordinate.getLongitude()};
+    }
+
+    @Override
+    protected List<Attribute> init(AbstractDefinition inputDefinition,
+                                   ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                                   SiddhiAppContext siddhiAppContext) {
+        initializeExtensionConfigs(configReader);
+        if (attributeExpressionExecutors.length != 1) {
+            throw new SiddhiAppValidationException("Invalid no of arguments passed to geo:geocoordinate(ip) " +
+                    "function, required 1, but found " + attributeExpressionExecutors.length);
+        }
+        Attribute.Type attributeType = attributeExpressionExecutors[0].getReturnType();
+        if (attributeType != Attribute.Type.STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for first argument ip of " +
+                    "geo:geocoordinate(ip) function, required " + Attribute.Type.STRING + ", but found " + attributeType
+                    .toString());
+        }
+        List<Attribute> attributes = new ArrayList<Attribute>(2);
+        attributes.add(new Attribute("latitude", Attribute.Type.DOUBLE));
+        attributes.add(new Attribute("longitude", Attribute.Type.DOUBLE));
+        return attributes;
+    }
+
+    @Override
+    public void start() {
+
+    }
+
+    @Override
+    public void stop() {
+
+    }
+
+    @Override
+    public Map<String, Object> currentState() {
+        return null;
+    }
+
+    @Override
+    public void restoreState(Map<String, Object> state) {
+
+    }
+
+    private void initializeExtensionConfigs(ConfigReader configReader)
+            throws SiddhiAppCreationException, SiddhiAppValidationException {
+        String geoResolverImplClassName = configReader.readConfig("geoCoordinateResolverClass",
+                DEFAULT_GEOCOORDINATE_RESOLVER_CLASSNAME);
+        try {
+            geoCoordinateResolverImpl = GeoCoordinateResolverHolder.
+                    getGeoCoordinationResolverHolderInstance().getGeoCoordinateResolver(geoResolverImplClassName);
+            geoCoordinateResolverImpl.init(configReader);
+        } catch (InstantiationException e) {
+            throw new SiddhiAppValidationException("Cannot instantiate GeoCoordinateResolverHolder holder class '"
```

Comment:
the closing quotation is missing in lines 156, 159, 162

Reply:
Fixed https://github.com/wso2-extensions/siddhi-execution-geo/pull/61/commits/a08d5e724a57f053d7d18ae441ace09bafeb208a